felix/routetable/route_table.go

```diff
@@ -1470,18 +1480,33 @@ func (r *RouteTable) applyUpdates(attempt int) error {
 
 	arpErrs := map[string]error{}
 	for ifaceName, addrToMAC := range r.pendingARPs {
-		// Add static ARP entries (for workload endpoints).  This may have been
-		// needed at one point but it no longer seems to be required.  Leaving
-		// it here for two reasons: (1) there may be an obscure scenario where
-		// it is needed. (2) we have tests that monitor netlink, and they break
-		// if it is removed because they see the ARP traffic.
+		if !r.ifacesToARP.Contains(ifaceName) {
```

Comment:
OK, right, ifacesToARP is the list of devices that needs to refresh ARP programming and pendingARPs is caching the ARPs per interfaces.

`for ifaceName, addrToMAC := range r.pendingARPs` could iterate over the ifacesToARP and then get the ARPs for that interface from pending.

Reply:
Yes, but I don't think that would be a clear win, because I'd need to check if there's an entry for ifaceName in r.pendingARPs.  So I think the code would still be equally complex.